{
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "I have a file abc.pyx that starts:\r\n```\r\n# distutils: language=c++\r\n```\r\n(because I am using vector).\r\n\r\nThis file compiled fine until I upgraded to Xcode 10.1 (10B61). I now get the following error:\r\n\r\n```\r\nwarning: include path for stdlibc++ headers not found; pass '-std=libc++' on the command line to use the libc++ standard\r\n      library instead [-Wstdlibcxx-not-found]\r\nabc.cpp:636:10: fatal error: 'ios' file not found\r\n#include \"ios\"\r\n         ^~~~~\r\n1 warning and 1 error generated.\r\nerror: command 'gcc' failed with exit status 1\r\n```\r\n\r\nAll of my pyx files that are compiled using C instead of C++ are compiling correctly.\r\n\r\nI am using Cython 0.29 in an Anaconda environment with Python 3.6.2 on macOS Mojave 10.14.1.\r\n```\r\nPython 3.6.2 |Continuum Analytics, Inc.| (default, Jul 20 2017, 13:14:59) \r\n[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin\r\n```\r\n\r\nMy setup.py looks like:\r\n```\r\nsetup(ext_modules=cythonize[(abc.pyx)])\r\n```\r\n\r\nI am building using the following command:\r\n```\r\npython setup.py build_ext --inplace\r\npython setup.py install\r\n```\r\n\r\nAny suggestions would be greatly appreciated. Is there any way I can pass a compiler option to just those files?\r\n\r\nThis [StackOverflow response](https://stackoverflow.com/a/40031250) is close, but I don't want to compile all of my pyx files into C++. I tried that, but got other errors. It would be simplest if I could just resolve the issue of this single file.",
    "closed_at": "2018-11-02T13:19:49Z",
    "comment_data": [
        {
            "author_association": "NONE",
            "body": "I followed the response to this [Stack Overflow question](https://stackoverflow.com/questions/52509602/cant-compile-c-program-on-a-mac-after-upgrade-to-mojave), but was still unable to compile.\r\n\r\nI have also determined the following:\r\n\r\n- The issue happens regardless of whether `--inplace` is used or not\r\n- If I run the Cython generated command manually at the command line, I have no errors\r\n- I have verified that the compiler version and python version are the same when running setup.py or when running the compilation command manually",
            "created_at": "2018-11-01T14:24:36Z",
            "html_url": "https://github.com/cython/cython/issues/2694#issuecomment-435056578",
            "id": 435056578,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2694",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQzNTA1NjU3OA==",
            "updated_at": "2018-11-01T14:24:36Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/435056578",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/422233?v=4",
                "events_url": "https://api.github.com/users/jsawruk/events{/privacy}",
                "followers_url": "https://api.github.com/users/jsawruk/followers",
                "following_url": "https://api.github.com/users/jsawruk/following{/other_user}",
                "gists_url": "https://api.github.com/users/jsawruk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/jsawruk",
                "id": 422233,
                "login": "jsawruk",
                "node_id": "MDQ6VXNlcjQyMjIzMw==",
                "organizations_url": "https://api.github.com/users/jsawruk/orgs",
                "received_events_url": "https://api.github.com/users/jsawruk/received_events",
                "repos_url": "https://api.github.com/users/jsawruk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/jsawruk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/jsawruk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/jsawruk"
            }
        },
        {
            "author_association": "NONE",
            "body": "This issue was resolved by upgrading to the Xcode Command Line Tools 10.1, released on November 2, 2018.",
            "created_at": "2018-11-02T13:19:49Z",
            "html_url": "https://github.com/cython/cython/issues/2694#issuecomment-435377772",
            "id": 435377772,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/2694",
            "node_id": "MDEyOklzc3VlQ29tbWVudDQzNTM3Nzc3Mg==",
            "updated_at": "2018-11-02T13:19:49Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/435377772",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/422233?v=4",
                "events_url": "https://api.github.com/users/jsawruk/events{/privacy}",
                "followers_url": "https://api.github.com/users/jsawruk/followers",
                "following_url": "https://api.github.com/users/jsawruk/following{/other_user}",
                "gists_url": "https://api.github.com/users/jsawruk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/jsawruk",
                "id": 422233,
                "login": "jsawruk",
                "node_id": "MDQ6VXNlcjQyMjIzMw==",
                "organizations_url": "https://api.github.com/users/jsawruk/orgs",
                "received_events_url": "https://api.github.com/users/jsawruk/received_events",
                "repos_url": "https://api.github.com/users/jsawruk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/jsawruk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/jsawruk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/jsawruk"
            }
        }
    ],
    "comments": 2,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/2694/comments",
    "created_at": "2018-10-31T16:48:01Z",
    "event_data": [
        {
            "actor": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/422233?v=4",
                "events_url": "https://api.github.com/users/jsawruk/events{/privacy}",
                "followers_url": "https://api.github.com/users/jsawruk/followers",
                "following_url": "https://api.github.com/users/jsawruk/following{/other_user}",
                "gists_url": "https://api.github.com/users/jsawruk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/jsawruk",
                "id": 422233,
                "login": "jsawruk",
                "node_id": "MDQ6VXNlcjQyMjIzMw==",
                "organizations_url": "https://api.github.com/users/jsawruk/orgs",
                "received_events_url": "https://api.github.com/users/jsawruk/received_events",
                "repos_url": "https://api.github.com/users/jsawruk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/jsawruk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/jsawruk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/jsawruk"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2018-11-02T13:19:49Z",
            "event": "closed",
            "id": 1942925402,
            "node_id": "MDExOkNsb3NlZEV2ZW50MTk0MjkyNTQwMg==",
            "url": "https://api.github.com/repos/cython/cython/issues/events/1942925402"
        }
    ],
    "events_url": "https://api.github.com/repos/cython/cython/issues/2694/events",
    "html_url": "https://github.com/cython/cython/issues/2694",
    "id": 376060030,
    "labels": [],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/2694/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "MDU6SXNzdWUzNzYwNjAwMzA=",
    "number": 2694,
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "closed",
    "title": "Xcode 10.1 error when compiling C++ files",
    "updated_at": "2018-11-02T13:19:49Z",
    "url": "https://api.github.com/repos/cython/cython/issues/2694",
    "user": {
        "avatar_url": "https://avatars1.githubusercontent.com/u/422233?v=4",
        "events_url": "https://api.github.com/users/jsawruk/events{/privacy}",
        "followers_url": "https://api.github.com/users/jsawruk/followers",
        "following_url": "https://api.github.com/users/jsawruk/following{/other_user}",
        "gists_url": "https://api.github.com/users/jsawruk/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/jsawruk",
        "id": 422233,
        "login": "jsawruk",
        "node_id": "MDQ6VXNlcjQyMjIzMw==",
        "organizations_url": "https://api.github.com/users/jsawruk/orgs",
        "received_events_url": "https://api.github.com/users/jsawruk/received_events",
        "repos_url": "https://api.github.com/users/jsawruk/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/jsawruk/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/jsawruk/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/jsawruk"
    }
}